src/main/scala/mesosphere/mesos/ResourceUtil.scala

```diff
@@ -1,121 +1,150 @@
-package mesosphere.marathon.tasks
+package mesosphere.mesos
 
 import com.twitter.util.NonFatal
+import mesosphere.marathon.state.DiskSource
+import org.apache.mesos.Protos.Resource.DiskInfo.Source
 import org.apache.mesos.Protos.Resource.{ DiskInfo, ReservationInfo }
 import org.apache.mesos.{ Protos => MesosProtos }
 import org.slf4j.LoggerFactory
 import scala.collection.JavaConverters._
 
 object ResourceUtil {
+  implicit class RichResource(resource: MesosProtos.Resource) {
+    def diskSourceOption: Option[Source] =
+      if (resource.hasDisk && resource.getDisk.hasSource)
+        Some(resource.getDisk.getSource)
+      else
+        None
+
+    def stringification: String = {
+      require(resource.getName == protos.Resource.DISK)
```

Comment:
question this `require` statement. seems less hazardous to implement as an `if` or `match` with a default case that uses toString.